File: Java(eclipse)/workspace/書籍検索アプリ/src/main/java/自作ライブラリ/ArrayContLib.java
package 自作ライブラリ;

import java.util.Arrays;

public class ArrayContLib {

	public static <T> T[] arrayConcat(T[] a, T[] b) {
			
		T[] result = Arrays.copyOf(a, a.length + b.length);
	    System.arraycopy(b, 0, result, a.length, b.length);
	    return result;
	}
}

File: 1_C#(VisualStudio)/1_workspace_ASP.NETCoreMVC/1_書籍検索アプリ_移植元Java/自作ライブラリ/ArrayContLib.cs
namespace 書籍検索アプリ_移植元Java.自作ライブラリ
{
    public static class ArrayContLib
    {
        public static T[] ArrayConcat<T>(T[] a, T[] b)
        {
            var result = new T[a.Length + b.Length];

            // a をコピー
            Array.Copy(a, 0, result, 0, a.Length);

            // b をコピー
            Array.Copy(b, 0, result, a.Length, b.Length);

            return result;
        }
    }
}